### PR TITLE
fix(multiselect): filter with virtualscroll enabled.

### LINKED ISSUE #7251: MultiSelect with filtering and virtual scroll enabled doesn't work properly.
**I'm submitting a ...**  (check one with "x")
```
[x] bug report => Search github for a similar issue or PR before submitting
[ ] feature request => Please check if request is not on the roadmap already https://github.com/primefaces/primeng/wiki/Roadmap
[ ] support request => Please do not submit support request here, instead see http://forum.primefaces.org/viewforum.php?f=35
```

**Plunkr Case (Bug Reports)**

https://stackblitz.com/edit/github-wpyzrc

**Current behavior**
In the `<p-multiSelect>`-element component you have created enabling the virtual scroll combined with filtering creates an issue, see the stackblitz link above.

When inserting a search in the filter field, the response only shows a result from the items currently inserted in the DOM.

**Expected behavior**
When enabling `[virtualScroll]="true"` and  `[filter]="true"` on a `<p-multiSelect>`-element, one would on filtering amongst all the values, expect a list of all matches in the multi-select-dropdown.

**Minimal reproduction of the problem with instructions**
Insert the following snippet of code and have at least some hundred values in the `options`-array.
```
<p-multiSelect
    [options]="options"
    [virtualScroll]="true"
    [panelStyle]="{ minWidth: '15em' }"
    [filter]="true"
    itemSize="30"
></p-multiSelect>
```

**What is the motivation / use case for changing the behavior?**

As a user would most likely appreciate all possible matching values when filtering on a set of values, this issue would imo need a fix.

**Please tell us about your environment:**

Windows, Visual Studio Code, yarn

* **Angular version:** 7.0.1

* **PrimeNG version:** 7.0.5

* **Browser:** Chrome 72.0.3626.96

 
* **Language:** [ TypeScript 3.1.4 | ES6 ]

### PR DESCRIPTION
**Pull request by PeterSkopal**
Change utilized array in p-multiSelectItem displaying actual filter results for users.

###Defect Fixes
#7251 

###Feature Requests
Due to company policy, we are unable to accept feature request PRs with significant changes as such cases has to be implemented by our team following our own processes.

### CODE REVIEW

File: src/app/showcase/components/multiselect/multiselectdemo.html [hunk 1]
```diff
@@ -27,7 +27,7 @@ <h3>Templating</h3>
     <p>Selected Cars: {{selectedCars2}}</p>
 
     <h3>Virtual Scroll (10000 Items)</h3>
-    <p-multiSelect [options]="items" [(ngModel)]="item" [panelStyle]="{minWidth:'12em'}" [virtualScroll]="true" [itemSize]="34" [filter]="false"></p-multiSelect>
+    <p-multiSelect [options]="items" [(ngModel)]="item" [panelStyle]="{minWidth:'12em'}" [virtualScroll]="true" [filter]="true" [itemSize]="34" [filter]="false"></p-multiSelect>
```
Comment (Mysame) [inline src/app/showcase/components/multiselect/multiselectdemo.html]:
```
[filter]="true" [itemSize]="34" [filter]="false"
```
That seems odd.

Comment (PeterSkopal) [reply src/app/showcase/components/multiselect/multiselectdemo.html]: @Mysame  Agree, honest mistake, should we create a new issue for it?

First `filter` input currently overrides the second one.

Comment (Mysame) [reply src/app/showcase/components/multiselect/multiselectdemo.html]: You probably should, as of now it isn't clear and looks wrong (even if it functions correctly)

Comment (PeterSkopal) [reply src/app/showcase/components/multiselect/multiselectdemo.html]: #7339

**Pull request merged**